### DISCUSSION TITLE
Spring Security Principal

### DISCUSSION THREAD

**Discussion by jkupcho** [Netflix/dgs-framework#367] [Q&A]
I'm curious if I'm missing something or if it's not implemented within the framework. Traditionally, you can add annotated arguments to Spring MVC methods i.e.:

```
@GetMapping
public SomeObject someMethod(@AuthenticationPrincipal Jwt principal) {
  ...
}
```

This does not seem to be supported within the `DgsQuery` annotated methods. The workaround seems to be using the `SecurityContextHolder` class, but I'm curious if a more "Spring" approach such as the one above is available.

Thanks!

**Comment by nayeemzen** [comment]
Any idea when this might be supported @paulbakker ? Can likely be part of [DataFetcherInvoker::invokeDataFetcher](https://github.com/Netflix/dgs-framework/blob/cb762dae368583c6a4a0754b6a845f859d9f5b59/graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DataFetcherInvoker.kt#L56)
